src/containers/SearchPage.jsx

```diff
@@ -8,30 +8,60 @@ import { Col, Row } from 'react-bootstrap';
 import * as actions from '../actions/searchActionCreators';
 
 export class SearchPage extends React.Component {
+  constructor(props) {
+    super(props);
+    const params = new URLSearchParams(this.props.location.search);
+    this.searchTerm = params.get('term');
+  }
+
+  componentDidMount() {
+    if (this.props.location.search) {
+      this.props.actions.submitSearch(this.searchTerm);
+    }
+
+    if (this.props.history.action === 'PUSH') {
+      this.props.actions.resetSearch();
+    }
+  }
 
-  showAppInfo() {
-    return this.props.search.submitted ? false : true;
+  componentWillReceiveProps(nextProps) {
+    if (
+      (nextProps.location !== this.props.location) && (nextProps.location.search === '')
+    ) {
+      this.props.actions.resetSearch();
+    }
   }
 
   render() {
+    const { submitted, submittedSearchTerm } = this.props.search
+    const { sortedIds, data, fetching } = this.props.results
+    const { submitSearch, getAdditionalInfo } = this.props.actions
     return (
         <div className="SearchPage">
           <Row>
             <Col xs={12} md={8} mdOffset={2}>
             <h1 className="text-center">I want to contribute to...</h1>
-            <SearchBar onSubmit={this.props.actions.submitSearch} />
+            <SearchBar
+              // setting a different key will rerender the component thus
+              // clearing the value
+              key={Math.random()}
```

Comment:
It looks like that's still the functionality though right? What I've noticed so far is:

1. If you perform a search for react, then a search for redux, the search bar input value will go back to redux
2. If you use the back button in the browser the URL will change properly, but the search results won't (Same with the forward button in the browser)

Thoughts on these?